### ISSUE TITLE
zfs-dkms rpm requires a 6.3 kernel, but code seems to be allowed to compile and run with 6.4

### ISSUE THREAD

**Issue by edgan** [openzfs/zfs#15151]
I am trying to install `kernel-6.4.7-200.fc38.x86_64` via `dnf` on `Fedora 38`. I get the error below. The funny part is I am already running `6.4.4-200.fc38.x86_64`. The issue is the `zfs-dkms` package will take any `kernel-devel-6.3` that is installed.

```
cat /proc/version 
Linux version 6.4.4-200.fc38.x86_64 (mockbuild@2aaadfeb41d049ad87820b2b55248688) (gcc (GCC) 13.1.1 20230614 (Red Hat 13.1.1-4), GNU ld version 2.39-9.fc38) #1 SMP PREEMPT_DYNAMIC Wed Jul 19 16:32:49 UTC 2023

```

```
rpm -q kernel-devel
kernel-devel-6.3.12-200.fc38.x86_64
kernel-devel-6.4.4-200.fc38.x86_64
kernel-devel-6.4.6-200.fc38.x86_64
```

```
dnf update
Last metadata expiration check: 0:15:32 ago on Thu 03 Aug 2023 02:47:27 PM MST.
Error: 
 Problem: package zfs-dkms-2.1.12-1.fc38.noarch from @System requires kernel-devel <= 6.3.999, but none of the providers can be installed
  - conflicting requests
  - problem with installed package zfs-dkms-2.1.12-1.fc38.noarch
(try to add '--skip-broken' to skip uninstallable packages)
```

**Comment by edgan** [comment]
I recompiled `zfs-dkms-2.1.12-1.fc38.src.rpm` after changing the spec file to define `6.4.99` instead of `6.3.99`. It works.

**Comment by tomchiverton** [comment]
This morning, dnf automatically removed zfs :(

/var/log/dnf.log
```
2023-08-11T06:01:00+0100 DEBUG Started dnf-automatic.
...
2023-08-11T06:05:48+0100 DEBUG Installed: kernel-6.4.9-200.fc38.x86_64
...
2023-08-11T06:05:48+0100 DEBUG Removed: zfs-2.1.12-1.fc38.x86_64
2023-08-11T06:05:48+0100 DEBUG Removed: zfs-dkms-2.1.12-1.fc38.noarch
```

manual fix
```
# yum install zfs zfs-dkms-2.1.12-1.fc38.noarch
Last metadata expiration check: 1:24:36 ago on Fri 11 Aug 2023 07:13:26 BST.
Dependencies resolved.
====================================================================================================================
 Package                      Architecture           Version                         Repository                Size
====================================================================================================================
Installing:
 kernel-devel                 x86_64                 6.2.9-300.fc38                  fedora                    19 M
 zfs                          x86_64                 2.1.12-1.fc38                   zfs                      662 k
 zfs-dkms                     noarch                 2.1.12-1.fc38                   zfs                       30 M
Removing:
 kernel-devel                 x86_64                 6.4.4-200.fc38                  @updates                  68 M
```

**Comment by adavidzh** [comment]
FWIW, I confirm that in Alma 9 and DKMS zfs 2.1.12:
- compiles and works ok with kernel 6.4.10, and
- does not compile with kernel 6.5.1.

**Comment by edgan** [comment]
This has been resolved by the new 2.2.0 release.
